recipients_app/lib/data/models/social_income_transaction.dart

```diff
@@ -109,10 +109,11 @@ class SocialIncomeTransaction extends Equatable {
     );
   }
 
-  factory SocialIncomeTransaction.fromJson(String source) =>
-      SocialIncomeTransaction.fromMap(
-        jsonDecode(source) as Map<String, dynamic>,
-      );
+  // TODO: do we need this? 
```

Comment:
normally we would need this to decode the string we are getting from firebase to a map

Reply:
So for getting object from firebase response I'm using `fromMap` directly (and passing id here as separate argument). `fromJson` was not used at all for now